server/src/main/java/com/linecorp/centraldogma/server/internal/httpapi/RepositoryServiceV1.java

```diff
@@ -0,0 +1,179 @@
+            throw new HttpResponseException(HttpStatus.BAD_REQUEST);
+        }
+        checkArgument(jsonNode != null, "repository name should be non-null");
+        final String repoName = jsonNode.asText();
+        checkArgument(!isNullOrEmpty(repoName), "repository name should be non-null");
+
+        return execute(Command.createRepository(System.currentTimeMillis(), AuthenticationUtil.currentAuthor(),
+                                                projectName, repoName))
+                .handle((unused, thrown) -> {
+                    if (thrown != null && thrown.getCause() instanceof StorageExistsException) {
+                        throw new HttpResponseException(HttpStatus.BAD_REQUEST);
+                    }
+                    return DtoConverter.convert(projectManager().get(projectName).repos().get(repoName));
+                });
+    }
+
+    /**
+     * DELETE /projects/{projectName}/repos/{repoName}
+     *
+     * <p>Remove a repository.
+     */
+    @Delete("/projects/{projectName}/repos/{repoName}")
+    public CompletionStage<Object> removeRepository(@Param("projectName") String projectName,
+                                                      @Param("repoName") String repoName) {
+        checkProjectExists(projectName);
+        final long removeTimeMillis = System.currentTimeMillis();
+        return execute(Command.removeRepository(removeTimeMillis, AuthenticationUtil.currentAuthor(),
+                                                projectName, repoName))
+                .handle((unused, thrown) -> {
+                    if (thrown != null && thrown.getCause() instanceof StorageNotFoundException) {
+                        throw new HttpResponseException(HttpStatus.BAD_REQUEST);
+                    }
+                    return VOID;
+                });
+    }
+
+    /**
+     * PATCH /projects/{projectName}/repos/{repoName}
+     *
+     * <p>Patches a repository with the JSON_PATCH. Currently, only unremove repository operation is supported.
+     */
+    @ConsumeType("application/json-patch+json")
+    @Patch("/projects/{projectName}/repos/{repoName}")
+    public CompletionStage<RepositoryDto> patchRepository(@Param("projectName") String projectName,
+                                                          @Param("repoName") String repoName,
+                                                          AggregatedHttpMessage message) {
+        checkProjectExists(projectName);
+
+        final JsonNode jsonNode;
+        try {
+            jsonNode = Jackson.readTree(message.content().toStringUtf8());
+        } catch (IOException e) {
+            throw new HttpResponseException(HttpStatus.BAD_REQUEST);
+        }
+
+        // Currently, only the unremoveRepository patch is supported.
+        if (!jsonNode.equals(unremovePatch)) {
+            throw new HttpResponseException(HttpStatus.BAD_REQUEST);
+        }
+
+        return execute(Command.unremoveRepository(System.currentTimeMillis(),
```

Comment:
You can just use the factory method without timestamp parameter to let the upstream decide.

Reply:
I was planning to use the value to return the client at first. I will remove them for now, and bring it back if it needs. :-)